website/server/controllers/api-v3/hall.js

```diff
@@ -164,17 +164,29 @@ api.getHero = {
   async handler (req, res) {
     let heroId = req.params.heroId;
 
-    req.checkParams('heroId', res.t('heroIdRequired')).notEmpty().isUUID();
+    req.checkParams('heroId', res.t('heroIdRequired')).notEmpty();
 
     let validationErrors = req.validationErrors();
     if (validationErrors) throw validationErrors;
 
     let hero = await User
-      .findById(heroId)
+      .findOne({'auth.local.username': heroId})
       .select(heroAdminFields)
       .exec();
 
-    if (!hero) throw new NotFound(res.t('userWithIDNotFound', {userId: heroId}));
+    if(!hero) {
+      req.checkParams('heroId', res.t('heroIdRequired')).notEmpty().isUUID();
```

Comment:
The `notEmpty` check here isn't necessary since you've done one above.

However you might be able to use the `isUUID` function to simplify the code. If the specified search term is a UUID, you can perform one type of search, otherwise you perform the other type.  You'd then need one `if(!hero)` section after both searches and you'd need only one set of `validationErrors` lines instead of two.